src/components/network/use-spreadsheet-equipments.js

```diff
@@ -5,25 +5,115 @@
-    );
+    const allEquipments = useSelector((state) => state.spreadsheetNetwork);
+    const equipments = allEquipments[equipment.type];
+
     const studyUuid = useSelector((state) => state.studyUuid);
     const currentNode = useSelector((state) => state.currentTreeNode);
     const [errorMessage, setErrorMessage] = useState();
     const [isFetching, setIsFetching] = useState(false);
 
+    const {
+        impactedSubstationsIds,
+        deletedEquipments,
+        impactedElementTypes,
+        resetImpactedSubstationsIds,
+        resetDeletedEquipments,
+        resetImpactedElementTypes,
+    } = useUpdateStudyImpacts();
+
     const shouldFetchEquipments = !equipments;
 
+    useEffect(() => {
+        // updating data related to impacted elements
+
+        // If we dont have any data in the spreadsheet, we don't need to update the equipments
+        const hasSpreadsheedData = () => {
+            return Object.values(allEquipments).some(
+                (value) => Array.isArray(value) && value.length > 0
+            );
+        };
+        if (!hasSpreadsheedData()) {
+            resetImpactedSubstationsIds();
+            resetDeletedEquipments();
+            resetImpactedElementTypes();
+            return;
+        }
+        // Handle updates and resets based on impacted element types
+        if (impactedElementTypes.length > 0) {
+            if (impactedElementTypes.includes(EQUIPMENT_TYPES.SUBSTATION)) {
+                dispatch(resetEquipments());
+                resetImpactedElementTypes();
+                return;
+            }
+            const impactedSpreadsheetEquipmentsTypes =
+                impactedElementTypes.filter((type) =>
+                    Object.keys(allEquipments).includes(type)
+                );
+            if (impactedSpreadsheetEquipmentsTypes.length > 0) {
+                dispatch(
+                    resetEquipmentsByTypes(impactedSpreadsheetEquipmentsTypes)
+                );
+            }
+            resetImpactedElementTypes();
+        }
+        if (impactedSubstationsIds.length > 0) {
+            // The formatting of the fetched equipments is done in the reducer
+            fetchAllEquipments(
+                studyUuid,
+                currentNode.id,
+                impactedSubstationsIds
+            ).then((values) => {
+                dispatch(updateEquipments(values));
+            });
+            resetImpactedSubstationsIds();
+        }
+        if (deletedEquipments.length > 0) {
+            deletedEquipments.forEach(({ equipmentType, equipmentId }) => {
+                if (equipmentId && equipmentType) {
+                    console.info(
+                        'removing equipment with id=',
+                        equipmentId,
+                        ' and type=',
+                        equipmentType,
+                        ' from the network'
+                    );
+                    dispatch(deleteEquipment(equipmentType, equipmentId));
```

Comment:
Maybe Add a deleteEquipments in redux action to mange all `deletedEquipments` into one dispatch